happi/client.py

```diff
@@ -526,6 +565,7 @@ def search_range(self, key, start, end=None, **kwargs):
                                               beamline='HXR')
         """
 
+        self._maybe_clear_cache()
```

Comment:
Is there ever a situation where we may need to clear the cache at the end of a search too?

Reply:
I think because the user intent is specified _prior_ to making a request that could hit the backend, the answer is "probably no". I can't think of a scenario within the given API where that could be the case, at least.